packages/pluggableWidgets/rich-text-web/src/utils/ckeditorPresets.ts

```diff
@@ -0,0 +1,156 @@
+            "-",
+            "CopyFormatting",
+            "RemoveFormat"
+        ]
+    },
+    {
+        name: "paragraph",
+        items: [
+            "NumberedList",
+            "BulletedList",
+            "-",
+            "Outdent",
+            "Indent",
+            "-",
+            "Blockquote",
+            "CreateDiv",
+            "-",
+            "JustifyLeft",
+            "JustifyCenter",
+            "JustifyRight",
+            "JustifyBlock",
+            "-",
+            "BidiLtr",
+            "BidiRtl",
+            "Language"
+        ]
+    },
+    { name: "links", items: ["Link", "Unlink", "Anchor"] },
+    { name: "insert", items: ["Image", "Table", "HorizontalRule", "Smiley", "SpecialChar", "PageBreak", "Iframe"] },
+    { name: "styles", items: ["Styles", "Format", "Font", "FontSize"] },
+    { name: "colors", items: ["TextColor", "BGColor"] },
+    { name: "tools", items: ["Maximize", "ShowBlocks"] },
+    { name: "about", items: ["About"] }
+];
+
+function SET_PRESET(type: "basic" | "standard" | "full"): CKEditorConfig {
+    const config: CKEditorConfig = {};
+    let toolbarGroup: Array<ToolbarGroup | string> = [];
+    switch (type) {
+        case "basic":
+            config.toolbarGroups = [...TOOLBAR_GROUP];
+            config.removeButtons =
+                "Source,Save,Templates,NewPage,ExportPdf,Preview,Print,Cut,Redo,Copy,Undo,Paste,PasteText,PasteFromWord,Find,Replace,SelectAll,Scayt,Form,Checkbox,Radio,TextField,Textarea,Select,Button,ImageButton,HiddenField,Strike,Underline,Subscript,Superscript,CopyFormatting,RemoveFormat,CreateDiv,Blockquote,JustifyLeft,BidiLtr,Image,Table,BidiRtl,JustifyCenter,JustifyRight,Language,JustifyBlock,HorizontalRule,SpecialChar,Smiley,PageBreak,Iframe,Styles,TextColor,BGColor,ShowBlocks,Maximize,Format,Font,FontSize,Anchor";
+            break;
+        case "standard":
+            toolbarGroup = [...TOOLBAR_GROUP];
+            toolbarGroup.splice(4, 0, "/");
+            config.toolbarGroups = toolbarGroup;
+            config.removeButtons =
+                "Save,Templates,NewPage,ExportPdf,Preview,Print,Find,Replace,SelectAll,Form,Checkbox,Radio,TextField,Textarea,Select,Button,ImageButton,HiddenField,Underline,Subscript,Superscript,CopyFormatting,CreateDiv,JustifyLeft,BidiLtr,BidiRtl,JustifyCenter,JustifyRight,Language,JustifyBlock,Smiley,PageBreak,Iframe,TextColor,BGColor,Font,FontSize,ShowBlocks";
+            break;
+        case "full":
+            toolbarGroup = [...TOOLBAR_GROUP];
+            toolbarGroup.splice(3, 0, "/");
+            toolbarGroup.splice(5, 0, "/");
+            toolbarGroup.splice(6, 0, "/");
+            config.toolbarGroups = toolbarGroup;
+            config.removeButtons = "";
+            /* temporary removed exportpdf*/
+            config.extraPlugins =
+                "save,templates,newpage,print,forms,find,selectall,copyformatting,div,divarea,justify,bidi,language,font,colorbutton,showblocks";
```

Comment:
Is this suddenly lowercase compared to the other switch cases intentional?

Reply:
I didn't understand the question, but the case is defined by CKEditor

Reply:
In L94 and L101 everything is camelcased: "Save", "Source", "SelectAll", but here it's just lowercased "save" "source"  "selectall". That seems a bit strange.

Reply:
Again, it's not controlled by me, it's ckeditor.
 CamelCased items are definition for toolbar group (Buttons). lowercase items are plugin names. They are matching but it's case sensitive.
 https://github.com/ckeditor/ckeditor4/blob/master/plugins/about/plugin.js check the link, it's an example of one of the plugins in ckeditor

Reply:
Okay, it wasn't clear that this is about plugins and the other cases are about removeButtons as the code is very similar 😓  if you don't think there's a more structured way to implement this, let's keep it like this i guess